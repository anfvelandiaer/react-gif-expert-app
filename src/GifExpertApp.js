import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     // No usar un push
    //     // categories.push('Hunter x Hunter)
    //     // Si usar:
    //     // setCategories([...categories, 'Hunter x Hunter']);
    //     setCategories(categories => [...categories, 'Hunter x Hunter']);
    // }
    
    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
            
            <ol>
                { 
                    categories.map(category => {
                        return <GifGrid 
                            key={category}
                            category={category}/>
                    })}
            </ol>
        </div>
    )
}
