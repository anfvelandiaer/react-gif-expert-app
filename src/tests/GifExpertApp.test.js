import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas en gif expert app component", () => {
  test("should mostrar el componente correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should mostrar el componente correctamente", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
