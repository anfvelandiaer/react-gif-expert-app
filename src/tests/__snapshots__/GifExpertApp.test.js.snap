// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gif expert app component should mostrar el componente correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
